bench/BenchAll.hs

```diff
@@ -416,6 +416,8 @@ main = do
           nf (S.foldl' (\acc x -> acc + fromIntegral x) (0 :: Int)) s) foldInputs
       , bgroup "foldr'" $ map (\s -> bench (show $ S.length s) $
           nf (S.foldr' (\x acc -> fromIntegral x + acc) (0 :: Int)) s) foldInputs
+      , bgroup "unfoldrN" $ map (\s -> bench (show $ S.length s) $
+          nf (S.unfoldrN (S.length s) (\_ -> Nothing) ) s) foldInputs
```

Comment:
This is a strange choice of benchmark: you are essentially building an empty string and do not invoke the inner loop of `unfoldrN` at all. Please put something like 
```haskell
nf (S.unfoldrN (S.length s) (\a -> (a, a + 1)) 0)
```